schemas/2016-03-01/Microsoft.ServiceFabric.json

```diff
@@ -0,0 +1,611 @@
+            },
+            "HttpApplicationGatewayCertificate": {
+              "oneOf": [
+                {
+                  "$ref": "#/definitions/CertificateDescription"
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ],
+              "description": "Microsoft.ServiceFabric/clusters: The server certificate used by http application gateway."
+            },
+            "AzureActiveDirectory": {
+              "oneOf": [
+                {
+                  "$ref": "#/definitions/AzureActiveDirectory"
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ],
+              "description": "Microsoft.ServiceFabric/clusters: The settings to enable AAD authentication on the cluster."
+            },
+            "ReliabilityLevel": {
+              "oneOf": [
+                {
+                  "enum": [
+                    "Invalid",
+                    "Bronze",
+                    "Silver",
+                    "Gold",
+                    "Platinum"
+                  ]
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ],
+              "description": "Microsoft.ServiceFabric/clusters: This level is used to set the number of replicas of the system services. Details: http://azure.microsoft.com/en-us/documentation/articles/service-fabric-cluster-capacity/#the-reliability-characteristics-of-the-cluster"
+            },
+            "NodeTypes": {
+              "oneOf": [
+                {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/definitions/NodeTypes"
+                  },
+                  "minItems": 1                  
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ],              
+              "description": "Microsoft.ServiceFabric/clusters: The node types of the cluster. Details: http://azure.microsoft.com/en-us/documentation/articles/service-fabric-cluster-capacity"
+            },
+            "ManagementEndpoint": {
+              "type": "string",
+              "minLength": 1,
+              "description": "Microsoft.ServiceFabric/clusters: The http management endpoint of the cluster."
+            },
+            "Certificate": {
+              "oneOf": [
+                {
+                  "$ref": "#/definitions/CertificateDescription"
```

Comment:
The definition of CertificateDescription also allows an expression. This would cause a problem because both rules of the "oneOf" at line 87 would apply if I actually give "Certificate" an expression value.